src/sentry/api/bases/external_actor.py

```diff
@@ -40,6 +47,17 @@ def validate_integration_id(self, integration_id: str) -> str:
     def validate_external_id(self, external_id: str) -> Optional[str]:
         return validate_external_id_option(external_id)
 
+    def validate_external_name(self, external_name: str) -> str:
```

Comment:
Since we don't want to force this restriction on other providers (like Slack), we first check to see if the provider is within a set of 'strict' providers. I reused the `validate_provider` code, but swallow the error in case they aren't a valid provider. This should be okay since the `validate_provider` will report its own failure when it runs, and `validate_external_name` should only report on the `external_name` field.

Reply:
In general, I try to only put one line per try/catch. For example if we refactored `validate_external_name()` to raise a `ParameterValidationError` then this function would never raise. I'd either move the call out of the try/catch:
```python
try:
    # Only validate the External Name if the provider is strict
    # (i.e. Users/Teams from Github or Gitlab)
    validate_provider(
        self.initial_data["provider"], available_providers=STRICT_NAME_PROVIDERS
    )
except (ParameterValidationError, KeyError):
    # Requests with invalid/absent providers will get handled by `validate_provider`
    return external_name
return validate_external_name(external_name)
```
or make a small helper that makes the flow clearer:
```python
def is_valid_provider(provider: str, available_providers: Optional[Set[ExternalProviders]] = None) -> bool:
    try:
        validate_provider(provider, available_providers)
    except:
        return False
    return True
```
```python    
def validate_external_name(self, external_name: str) -> str:
    provider = validate_provider(self.initial_data.get("provider"))
    if is_valid_provider(provider):
        return validate_external_name(external_name)
    return external_name    
```

Reply:
This is really smart, thanks for pointing it out! Added in the 1585365 commit.